File: python/mailpro_client.py
import os
import requests

class MailproClient:
    def __init__(self, token=None, base_url='https://api.mailpro.com'):
        self.token = token or os.getenv('MAILPRO_TOKEN')
        if not self.token:
            raise ValueError('MAILPRO_TOKEN required')
        self.base_url = base_url.rstrip('/')
        self.headers = {
            'Authorization': f'Bearer {self.token}',
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        }

    def _url(self, path):
        return f"{self.base_url}{path}"

    def post(self, path, payload=None):
        r = requests.post(self._url(path), headers=self.headers, json=payload)
        r.raise_for_status()
        return r.json()

    def get(self, path, params=None):
        r = requests.get(self._url(path), headers=self.headers, params=params)
        r.raise_for_status()
        return r.json()

    def put(self, path, payload=None):
        r = requests.put(self._url(path), headers=self.headers, json=payload)
        r.raise_for_status()
        return r.json()

    # Email
    def send_email(self, id_from, to, subject, html, **kwargs):
        payload = dict(idFrom=id_from, to=to, subject=subject, bodyHtml=html)
        payload.update(kwargs)
        return self.post('/v3/email/send', payload)

    # Contacts
    def create_contact(self, payload):
        return self.post('/v3/contacts', payload)

    def update_contact(self, contact_id, payload):
        return self.put(f'/v3/contacts/{contact_id}', payload)

    def delete_contact(self, contact_id):
        r = requests.delete(self._url(f'/v3/contacts/{contact_id}'), headers=self.headers)
        r.raise_for_status()
        return r.json() if r.text else {}

    # Campaigns
    def create_campaign(self, payload):
        return self.post('/v3/campaigns', payload)

    def get_campaign(self, campaign_id):
        return self.get(f'/v3/campaigns/{campaign_id}')

    def send_campaign(self, campaign_id):
        return self.post(f'/v3/campaigns/{campaign_id}/send')

File: nodejs/src/mailproClient.js
const fetch = require('node-fetch');

class MailproClient {
  constructor(apiKey) {
    this.apiKey = apiKey;
    this.baseUrl = 'https://api.mailpro.com/v3';
  }

  async request(endpoint, method = 'GET', body = null) {
    const res = await fetch(`${this.baseUrl}${endpoint}`, {
      method,
      headers: {
        'Authorization': `Bearer ${this.apiKey}`,
        'Content-Type': 'application/json'
      },
      body: body ? JSON.stringify(body) : null
    });
    if (!res.ok) {
      const errText = await res.text();
      throw new Error(`Mailpro API error: ${res.status} ${errText}`);
    }
    return res.json();
  }

  sendEmail(payload) {
    return this.request('/emails', 'POST', payload);
  }

  getContacts() {
    return this.request('/contacts', 'GET');
  }

  createContact(payload) {
    return this.request('/contacts', 'POST', payload);
  }

  createCampaign(payload) {
    return this.request('/campaigns', 'POST', payload);
  }

  getCampaign(id) {
    return this.request(`/campaigns/${id}`, 'GET');
  }
}

module.exports = MailproClient;
